src/main/java/tc/oc/pgm/match/MatchManagerImpl.java

```diff
@@ -159,6 +159,11 @@ public void unloadMatch(@Nullable String id) {
 
     matchIdByWorldName.remove(getWorldName(id));
     matchById.remove(id);
+    // Destroy the match 5 seconds after unload, to let the server catch-up
```

Comment:
Can we delay destroy until the server TPS recovers to a certain threshold? 5 seconds is arbitrary especially depending on machine type. Maybe it can wait a max of 1 minute?

Reply:
This is all just still testing, we really don't really know the impact of this. I want this to be tested/profiled before actually deciding what exactly we do with this